packages/react/src/components/Provider/index.tsx

```diff
@@ -0,0 +1,86 @@
+import React, {
+  createContext,
+  PropsWithChildren,
+  useContext,
+  useMemo,
+} from "react";
+import * as enUS from "locales/en-US.json";
+import * as frFR from "locales/fr-FR.json";
+import { PartialNested } from "types";
+
+type TranslationSet = PartialNested<typeof enUS>;
+
+const CunninghamContext = createContext<
+  | undefined
+  | {
+      t: (key: string, vars?: Record<string, string | number>) => string;
+    }
+>(undefined);
+
+export const useCunningham = () => {
+  const context = useContext(CunninghamContext);
+  if (context === undefined) {
+    throw new Error("useCunningham must be used within a CunninghamProvider.");
+  }
+  return context;
+};
+
+interface Props extends PropsWithChildren {
+  customLocales?: Record<string, TranslationSet>;
+  currentLocale?: string;
+}
+
+export const SUPPORTED_LOCALES = ["en-US", "fr-FR"];
+
+export const CunninghamProvider = ({
+  currentLocale = "en-US",
+  customLocales,
+  children,
+}: Props) => {
+  const locales: Record<string, TranslationSet> = useMemo(
+    () => ({
+      "en-US": enUS,
+      "fr-FR": frFR,
+      ...customLocales,
+    }),
+    [customLocales]
+  );
+
+  if (!locales[currentLocale]) {
+    throw new Error(
+      `CunninghamProvider: ${currentLocale} is not supported. Please provide a translation set via customLocales prop.`
+    );
+  }
+
+  const context = useMemo(
+    () => ({
+      t: (key: string, vars?: Record<string, string | number>) => {
+        const [namespace, ...keys] = key.split(".");
+        let message = keys.reduce(
+          (acc, subKey) => acc[subKey],
+          (locales[currentLocale] as any)[namespace]
+        );
+
+        if (!message) {
+          throw new Error(`Translation not found for key: ${key}`);
```

Comment:
Same here, if we consider using a default locale, can we use the messages from this default locale ?